app/src/pages/inside/logsPage/historyLine/historyLine.jsx

```diff
@@ -39,33 +55,68 @@ const cx = classNames.bind(styles);
   },
   { namespace: NAMESPACE },
 )
-@connect((state) => ({
-  historyItems: historyItemsSelector(state),
-  activeItemId: activeLogIdSelector(state),
-}))
+@connect(
+  (state) => ({
+    historyItems: historyItemsSelector(state),
+    activeItemId: activeLogIdSelector(state),
+  }),
+  {
+    fetchHistoryItemsAction,
+  },
+)
+@injectIntl
 export class HistoryLine extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { isLoading: false };
+  }
   static propTypes = {
+    intl: PropTypes.object.isRequired,
     historyItems: PropTypes.array,
     activeItemId: PropTypes.number,
     changeActiveItem: PropTypes.func,
+    fetchHistoryItemsAction: PropTypes.func,
   };
 
   static defaultProps = {
     historyItems: [],
     activeItemId: 0,
     changeActiveItem: () => {},
+    fetchHistoryItemsAction: () => {},
   };
 
   checkIfTheItemLinkIsActive = (item) =>
     item.id !== this.props.activeItemId && item.status !== NOT_FOUND;
 
-  render() {
-    const { historyItems, activeItemId, changeActiveItem } = this.props;
+  setLoading = (isLoading = false) => this.setState({ isLoading });
 
+  loadMoreItems = () => {
+    this.setState({ isLoading: true });
+    this.props.fetchHistoryItemsAction({ loadMore: true, setLoading: this.setLoading });
+  };
+
+  render() {
+    const { historyItems, activeItemId, changeActiveItem, intl } = this.props;
+    const shouldShowLoadMore =
+      (historyItems.length !== 30 && historyItems.length === DEFAULT_HISTORY_DEPTH) ||
+      (historyItems.length - DEFAULT_HISTORY_DEPTH) % NUMBER_OF_ITEMS_TO_LOAD === 0;
```

Comment:
This condition not working.
The button still shown when we have f.e. 3 loaded items.

Reply:
Try this
```
    const loadedItems = historyItems.length - DEFAULT_HISTORY_DEPTH;
    const shouldShowLoadMore =
      historyItems.length < HISTORY_DEPTH_LIMIT &&
      loadedItems >= 0 &&
      loadedItems % NUMBER_OF_ITEMS_TO_LOAD === 0;
```